Add Twake and custom SSO

zatteo opened the pull request.

zatteo committed:
feat: Add a button to log in with Twake

This button opens a webview on a stack route that manage OIDC login.

zatteo committed:
feat: Manage OIDC login with Twake

When the user login to Twake SSO in webview, he's redirected to a deep link that we can intercept to get instance and oidcCode.

We can then use instance and oidcCode to login by calling a route from the stack.

zatteo committed:
feat: Listen to deep link that return context

For SaaS users, we will create a new route on the cloudery that ask
for email and return context and with this context, we can open the
stack route with the context.

Here we implement the deeplink.

zatteo committed:
feat: Login with company server with .well-known

zatteo committed:
feat: Login with company server with direct URL

zatteo committed:
feat: Improve company login UI

Crash-- commented on src/Core/Services/CozyClouderyEnvService.cs:
```diff
@@ -48,6 +53,22 @@ public async Task<string> GetClouderyUrl()
 
+        public async Task<string> GetStackOidcUrl(string context = "twake")
+        {
+            var clouderyEnv = await GetClouderyEnvFromAsyncStorage();
```
I don't remember how we setup this cloudery env? It is during the compilation phase?

Crash-- commented:
Looks great. Can we have a testflight or something to test it?

Crash-- commented on src/Core/Services/CozyClouderyEnvService.cs:
```diff
@@ -15,6 +15,11 @@ public class CozyClouderyEnvService: ICozyClouderyEnvService
         private const string DEV_BASE_URI = "https://manager-dev.cozycloud.cc";
 
+
+        private const string PROD_STACK_OAUTHCALLBACK_URI = "https://oauthcallback.mycozy.cloud";
+        private const string INT_STACK_OAUTHCALLBACK_URI = "https://oauthcallback.cozy.works";
+        private const string DEV_STACK_OAUTHCALLBACK_URI = "https://oauthcallback.cozy.wtf";
+
```
What I don't like with these urls, is that tomorrow, we should use oauthcallback.twake.app, lin-saas... 

So having the mycozy.cloud will result to make an extra http redirection at the end, but I don't see other way to do it right now to be honest.

zatteo committed:
feat: Translate login wording

zatteo committed:
fix: Set correct color for authentication button

zatteo committed:
fix: Send hash in OIDC login

During the SSO login, we still need to enter the master password even
if we can be considered as logged in (= having access tokens).

Before, we were never checking if the master password entered was
correct so the app was raising an crypto error
javax.crypto.BadPaddingException when trying to decipher the vault.
So here we send the hash password during the SSO login to be sure
that when we have our access tokens, the master password is also the
good one.

See https://github.com/cozy/cozy-stack/pull/4566
